meson.build

```diff
@@ -339,6 +339,11 @@ if posix
     flags += ['-D_GNU_SOURCE']
 endif
 
+if host_machine.system() == 'android'
+    flags += ['-Werror=unguarded-availability',
+              '-D__ANDROID_UNAVAILABLE_SYMBOLS_ARE_WEAK__']
```

Comment:
I am a bit worried about this inadvertently causing bugs to slip through but I guess with the Werror flag that should be fine.
Alternatively we could scope this define only to `ao_audiotrack.c`, where it's needed?